### PR TITLE
Making list large dir test parallel

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Made list large files package independent, to make it parallel run with other packages. We found 5-7 minutes improvement after this.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - Automated

### CODE REVIEW

File: perfmetrics/scripts/run_e2e_tests.sh [hunk 1]
```diff
@@ -53,28 +53,162 @@ sudo apt-get install -y gcc python3-dev python3-setuptools
...
+# Non parallel execution of integration tests located within specified test directories for group 1.
+function run_non_parallel_tests_group_1() {
+  for test_dir_np_group_1 in "${test_dir_non_parallel_group_1[@]}"
+  do
+    test_path_non_parallel_group_1="./tools/integration_tests/$test_dir_np_group_1"
+    # Executing integration tests
+    GODEBUG=asyncpreemptoff=1 go test $test_path_non_parallel_group_1 -p 1 --integrationTest -v --testbucket=$BUCKET_NAME_NON_PARALLEL_GROUP_1 --testInstalledPackage=$run_e2e_tests_on_package -timeout $INTEGRATION_TEST_TIMEOUT
+    exit_code_non_parallel_group_1=$?
+    if [ $exit_code_non_parallel_group_1 != 0 ]; then
+      test_fail_np_group_1=$exit_code_non_parallel_group_1
```
Comment (sethiay) [inline perfmetrics/scripts/run_e2e_tests.sh:77]: why can't we simply return exit_code_non_parallel_group_1?

File: perfmetrics/scripts/run_e2e_tests.sh [hunk 2]
```diff
@@ -53,28 +53,162 @@ sudo apt-get install -y gcc python3-dev python3-setuptools
...
+    fi
+  done
+  return $test_fail_p
+}
+
+# Test setup
+# Create Bucket for non parallel e2e tests
+# The bucket prefix for the random string
+bucketPrefix="gcsfuse-non-parallel-e2e-tests-group-1-"
+BUCKET_NAME=$(create_bucket $bucketPrefix)
```
Comment (sethiay) [inline perfmetrics/scripts/run_e2e_tests.sh:128]: we can directly assign to BUCKET_NAME_NON_PARALLEL_GROUP_1?

File: perfmetrics/scripts/run_e2e_tests.sh [hunk 3]
```diff
@@ -53,28 +53,162 @@ sudo apt-get install -y gcc python3-dev python3-setuptools
...
 set +e
-# Executing integration tests
-GODEBUG=asyncpreemptoff=1 go test ./tools/integration_tests/... -p 1 --integrationTest -v --testbucket=$BUCKET_NAME --testInstalledPackage=$run_e2e_tests_on_package -timeout $INTEGRATION_TEST_TIMEOUT
-exit_code=$?
 
+echo "Running parallel tests..."
+# Run parallel tests
+run_parallel_tests &
+my_process_p=$!
+echo "Running non
```
Comment (sethiay) [inline perfmetrics/scripts/run_e2e_tests.sh:183]: nit: you meant echo "Running non parallel tests group 1... "

**Review by sethiay — COMMENTED** [review]
Also, could you confirm that the gcsfuse debug logs of different tests would not interfere with each ?

File: perfmetrics/scripts/run_e2e_tests.sh (hunk 1 above)
Comment (Tulsishah) [reply perfmetrics/scripts/run_e2e_tests.sh:77]: If we return from inside the loop, the loop will not run for all array packages.

File: perfmetrics/scripts/run_e2e_tests.sh (hunk 2 above)
Comment (Tulsishah) [reply perfmetrics/scripts/run_e2e_tests.sh:128]: @ankitaluthra1 added this change in her PR.

File: perfmetrics/scripts/run_e2e_tests.sh (hunk 3 above)
Comment (Tulsishah) [reply perfmetrics/scripts/run_e2e_tests.sh:183]: You have to review this PR 
https://github.com/GoogleCloudPlatform/gcsfuse/pull/1775/files

This is draft only on top of above PR.

File: tools/integration_tests/list_large_dir/list_large_dir_test.go [hunk 4]
```diff
@@ -24,6 +24,9 @@ import (
 	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
 )
 
+const DirectoryForListLargeFileTests1 = "directoryForListLargeFileTests1"
```
Comment (ashmeenkaur) [inline tools/integration_tests/list_large_dir/list_large_dir_test.go:27]: IIUC, only the package is running the tests in prallel to other packages. Couldn't we use just 1 test directory?

**Review by ashmeenkaur — APPROVED** [review]

**Review by raj-prince — APPROVED** [review]

**Pull request merged**